src/repl-frontend/scala/tools/nsc/MainGenericRunner.scala

```diff
@@ -62,9 +62,9 @@ class MainGenericRunner {
         case AsObject =>
           ObjectRunner.runAndCatch(settings.classpathURLs, thingToRun, command.arguments)
         case AsScript if isE =>
-          Right(ScriptRunner.runCommand(settings, combinedCode, thingToRun +: command.arguments))
+          Right(!ScriptRunner(settings).runScriptText(combinedCode, thingToRun +: command.arguments).isDefined)
```

Comment:
why do we discard the exception here?

Reply:
You never let me get away with anything. I was worn down by the refactor at that point. And by figuring out what to call it besides `runCommand`. I'll take another look at how the values bubble up; there were other spots like `CommonRunner` IIRC.

Reply:
Perhaps along with the rebase?